src/Tools/dotnet-monitor/CollectionRules/Actions/CollectTraceAction.cs

```diff
@@ -42,11 +44,13 @@ private sealed class CollectTraceAction :
             CollectionRuleActionBase<CollectTraceOptions>
         {
             private readonly IServiceProvider _serviceProvider;
+            private readonly IOptionsMonitor<GlobalCounterOptions> _counterOptions;
 
             public CollectTraceAction(IServiceProvider serviceProvider, IEndpointInfo endpointInfo, CollectTraceOptions options)
                 : base(endpointInfo, options)
             {
                 _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+                _counterOptions = _serviceProvider.GetRequiredService<IOptionsMonitor<GlobalCounterOptions>>();
```

Comment:
This will need to be configured on the configuration builder in the TestHostHelper: https://github.com/dotnet/dotnet-monitor/blob/42b94c9b77f6c30b5641c2d6bc2819ab4b8646ed/src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/TestHostHelper.cs#L76